core/sql/generator/GenPreCode.cpp

```diff
@@ -170,12 +174,16 @@ static NABoolean processConstHBaseKeys(Generator * generator,
            skey->getIndexDesc()->getOrderOfKeyValues(),
            relExpr->getGroupAttr()->getCharacteristicInputs(),
            TRUE, /* forward scan */
-           exePreds,
+           keyPreds,
            nonKeyColumnSet,
            idesc,
            relExpr->getGroupAttr()->getCharacteristicOutputs(),
            mySearchKeys);
 
+      // Include any remaining key predicates that have not been 
+      // picked up (to be used as the HBase search keys).
+      exePreds += keyPreds;
```

Comment:
Yes. Please refer to the comment of the method. It contains the
"left-over".

On Thu, Aug 27, 2015 at 10:31 AM, DaveBirdsall <notifications@github.com>
wrote:

> In core/sql/generator/GenPreCode.cpp
> <https://github.com/apache/incubator-trafodion/pull/66#discussion_r38109234>
> :
>
> >             nonKeyColumnSet,
> >             idesc,
> >             relExpr->getGroupAttr()->getCharacteristicOutputs(),
> >             mySearchKeys);
> >
> > +      // Include any remaining key predicates that have not been
> > +      // picked up (to be used as the HBase search keys).
> > +      exePreds += keyPreds;
>
> So, is keyPreds side-effected by the HbaseSearchKey::makeHBaseSearchKeys
> call? Then it might be different when added back.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-trafodion/pull/66/files#r38109234>.
>



-- 
Regards, --Qifan